sql: remove EXPLAIN(DEBUG)

knz opened the pull request:
(now replaced by SHOW [SESSION] KV TRACE)

This in turn enables deleting the `MarkDebug` / `DebugValues`
interface and cleaning up the Next() code in most nodes.

Supersedes #16514.

Fixes #9563.
Fixes #15948.
Adds the missing test from #15925.

cc @andreimatei

knz committed:
sql: replace all EXPLAIN(DEBUG) tests to use SHOW KV TRACE FOR instead

knz committed:
sql: remove EXPLAIN(DEBUG)

(now replaced by SHOW [SESSION] KV TRACE)

This in turn enables deleting the `MarkDebug` / `DebugValues`
interface and cleaning up the Next() code in most nodes.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/16694)
<!-- Reviewable:end -->

knz committed:
sql: add the missing test from #15925

The fix committed in 0cf238335f7583bcee1d9a4017ee940fd9c0772e was
missing a test because EXPLAIN(DEBUG) wasn't able to exercise the
corrected code path. Now that we can exercise the path, do it.

andreimatei commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> :heart:

---

Review status: 0 of 62 files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[docs/RFCS/explain_debug.md, line 7 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/16694#-KnL68gSlsKA0C64a7Gf:-KnL68gSlsKA0C64a7Gg:b-ree8xv) ([raw file](https://github.com/cockroachdb/cockroach/blob/8bdb0b56574c80f7b60e2a5e26c35f33deead8b9/docs/RFCS/explain_debug.md#L7)):*
> ```Markdown
> - Cockroach Issue:
> 
> **Note: obsoleted in favor of session tracing.**
> ```

I think we should just delete the file.

---

*[pkg/sql/logictest/testdata/logic_test/select_index, line 605 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/16694#-KnL71kqGWg3quW5JB14:-KnL71kqGWg3quW5JB15:b-l7x5w7) ([raw file](https://github.com/cockroachdb/cockroach/blob/63def9ca872238952f9a0c72b50967df3eb4c015/pkg/sql/logictest/testdata/logic_test/select_index#L605)):*
> ```
>      VALUES ('001'),('002'),('003'),('004'),('005'),('006'),('007'),('008'),('009'),('010'),
>             ('011'),('012'),('013'),('014'),('015'),('016'),('017'),('018'),('019'),('020'),
> 	    ('021'),('022'),('023'),('024'),('025'),('026'),('027'),('028'),('029'),('030')
> ```

nit: weird formatting, there's a tab in there

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/16694#-:-KnL7CUb37UkO7dotYDr:bsk1ten)*
<!-- Sent from Reviewable.io -->

knz commented:
Review status: 0 of 62 files reviewed at latest revision, 2 unresolved discussions.

---

*[docs/RFCS/explain_debug.md, line 7 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/16694#-KnL68gSlsKA0C64a7Gf:-KnLerd9wygJUqfJSK6i:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/8bdb0b56574c80f7b60e2a5e26c35f33deead8b9/docs/RFCS/explain_debug.md#L7)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

I think we should just delete the file.
</blockquote></details>

Done.

---

*[pkg/sql/logictest/testdata/logic_test/select_index, line 605 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/16694#-KnL71kqGWg3quW5JB14:-KnLesZ9BLaOBsEFAqvW:bt2vr1b) ([raw file](https://github.com/cockroachdb/cockroach/blob/63def9ca872238952f9a0c72b50967df3eb4c015/pkg/sql/logictest/testdata/logic_test/select_index#L605)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

nit: weird formatting, there's a tab in there
</blockquote></details>

Fixed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/16694)*
<!-- Sent from Reviewable.io -->

knz commented:
TFYR!
